### ISSUE TITLE
WDT discoverDomain version is 3.0.1:.94c1f46 - Exception: exceptions.AttributeError

### ISSUE THREAD

**Issue by Michalski-Piotr** [oracle/weblogic-deploy-tooling#1415]
```console


[oracle@o2c-domain-as ~]$ /u01/oracle/scripts/weblogic-deploy/bin/discoverDomain.sh -oracle_home $FMW_HOME -domain_home  $DOMAIN_ROOT/$DOMAIN_NAME -admin_url t3://10.100.4.189:7101 -admin_user weblogic -model_file DiscoveredPATOCI.yaml -archive_file o2c.zip
JDK version is 1.8.0_361-b33
JAVA_HOME = /usr/java/latest
WLST_EXT_CLASSPATH = /u01/oracle/scripts/weblogic-deploy/lib/weblogic-deploy-core.jar
CLASSPATH = /u01/oracle/scripts/weblogic-deploy/lib/weblogic-deploy-core.jar
WLST_PROPERTIES = -Dcom.oracle.cie.script.throwException=true -Djava.util.logging.config.class=oracle.weblogic.deploy.logging.WLSDeployLoggingConfig
/u01/oracle/fmw/oracle_common/common/bin/wlst.sh /u01/oracle/scripts/weblogic-deploy/lib/python/discover.py -oracle_home /u01/oracle/fmw -domain_home /u01/oracle/domains/o2c-domain -admin_url t3://10.100.4.189:7101 -admin_user weblogic -model_file DiscoveredPATOCI.yaml -archive_file o2c.zip

Initializing WebLogic Scripting Tool (WLST) ...

Welcome to WebLogic Server Administration Scripting Shell

Type help() for help on available commands

####<Mar 13, 2023 12:48:39 PM> <INFO> <WebLogicDeployToolingVersion> <logVersionInfo> <WLSDPLY-01750> <The WebLogic Deploy Tooling discoverDomain version is 3.0.1:.94c1f46:Feb 23, 2023 14:43 UTC>
####<Mar 13, 2023 12:48:39 PM> <INFO> <WLSDeployLoggingConfig> <logLoggingDirectory> <WLSDPLY-01755> <The discoverDomain program will write its log to directory /u01/oracle/scripts/weblogic-deploy/logs>
Please enter the WebLogic administrator password:
####<Mar 13, 2023 12:48:43 PM> <INFO> <ModelContext> <__copy_from_args> <WLSDPLY-01050> <WebLogic version for aliases is 12.2.1.4.0.221210>
####<Mar 13, 2023 12:48:43 PM> <INFO> <discover> <main> <WLSDPLY-06024> <No variable file provided. Model passwords will contain the token '-- FIX ME --'>



####<Mar 13, 2023 12:49:40 PM> <INFO> <discover> <_get_domain_name> <WLSDPLY-06022> <Discover domain o2c-domain>
####<Mar 13, 2023 12:49:40 PM> <INFO> <Aliases> <set_production_mode> <WLSDPLY-19047> <Discovering domain in production mode>
####<Mar 13, 2023 12:49:42 PM> <INFO> <TopologyDiscoverer> <discover> <WLSDPLY-06600> <Discovering domain model topology>
####<Mar 13, 2023 12:49:42 PM> <INFO> <Discoverer> <discover_domain_mbean> <WLSDPLY-06644> <Adding Domain AdminConsole>
####<Mar 13, 2023 12:49:43 PM> <INFO> <Discoverer> <discover_domain_mbean> <WLSDPLY-06644> <Adding Domain CdiContainer>
####<Mar 13, 2023 12:49:43 PM> <INFO> <Discoverer> <discover_domain_mbean> <WLSDPLY-06644> <Adding Domain JMX>
####<Mar 13, 2023 12:49:43 PM> <INFO> <Discoverer> <discover_domain_mbean> <WLSDPLY-06644> <Adding Domain JPA>
####<Mar 13, 2023 12:49:43 PM> <INFO> <Discoverer> <discover_domain_mbean> <WLSDPLY-06644> <Adding Domain JTA>
####<Mar 13, 2023 12:49:44 PM> <INFO> <Discoverer> <discover_domain_mbean> <WLSDPLY-06644> <Adding Domain Log>
####<Mar 13, 2023 12:49:44 PM> <INFO> <Discoverer> <discover_domain_mbean> <WLSDPLY-06644> <Adding Domain RestfulManagementServices>
####<Mar 13, 2023 12:49:45 PM> <INFO> <TopologyDiscoverer> <get_clusters> <WLSDPLY-06601> <Discovering 1 clusters>
####<Mar 13, 2023 12:49:45 PM> <INFO> <TopologyDiscoverer> <get_clusters> <WLSDPLY-06602> <Adding Cluster soa_cluster>
####<Mar 13, 2023 12:49:47 PM> <INFO> <TopologyDiscoverer> <get_servers> <WLSDPLY-06603> <Discovering 3 servers>
####<Mar 13, 2023 12:49:47 PM> <INFO> <TopologyDiscoverer> <get_servers> <WLSDPLY-06604> <Adding Server AdminServer>
####<Mar 13, 2023 12:50:07 PM> <INFO> <TopologyDiscoverer> <get_servers> <WLSDPLY-06604> <Adding Server soa_server1>
####<Mar 13, 2023 12:50:26 PM> <INFO> <TopologyDiscoverer> <get_servers> <WLSDPLY-06604> <Adding Server soa_server2>
####<Mar 13, 2023 12:50:43 PM> <INFO> <TopologyDiscoverer> <get_migratable_targets> <WLSDPLY-06607> <Discovering 2 Migratable Targets>
####<Mar 13, 2023 12:50:44 PM> <INFO> <TopologyDiscoverer> <get_migratable_targets> <WLSDPLY-06608> <Adding Migratable Target soa_server1 (migratable)>
####<Mar 13, 2023 12:50:44 PM> <INFO> <TopologyDiscoverer> <get_migratable_targets> <WLSDPLY-06608> <Adding Migratable Target soa_server2 (migratable)>
####<Mar 13, 2023 12:50:45 PM> <INFO> <TopologyDiscoverer> <get_server_templates> <WLSDPLY-06605> <Discovering 3 server templates>
####<Mar 13, 2023 12:50:45 PM> <INFO> <TopologyDiscoverer> <get_server_templates> <WLSDPLY-06606> <Adding soa-server-template Server Template>
####<Mar 13, 2023 12:51:02 PM> <INFO> <TopologyDiscoverer> <get_server_templates> <WLSDPLY-06606> <Adding wsm-cache-server-template Server Template>
####<Mar 13, 2023 12:51:19 PM> <INFO> <TopologyDiscoverer> <get_server_templates> <WLSDPLY-06606> <Adding wsmpm-server-template Server Template>
####<Mar 13, 2023 12:51:36 PM> <INFO> <TopologyDiscoverer> <get_unix_machines> <WLSDPLY-06609> <Discovering 2 Unix machines>
####<Mar 13, 2023 12:51:36 PM> <INFO> <TopologyDiscoverer> <get_machines> <WLSDPLY-06611> <Discovering 2 machines>
####<Mar 13, 2023 12:51:36 PM> <INFO> <TopologyDiscoverer> <get_machines> <WLSDPLY-06612> <Adding Machine machine1>
####<Mar 13, 2023 12:51:36 PM> <INFO> <TopologyDiscoverer> <get_machines> <WLSDPLY-06612> <Adding Machine machine2>
####<Mar 13, 2023 12:51:37 PM> <INFO> <TopologyDiscoverer> <discover_security_configuration> <WLSDPLY-06622> <Adding Security Configuration>
####<Mar 13, 2023 12:51:41 PM> <INFO> <TopologyDiscoverer> <get_embedded_ldap_configuration> <WLSDPLY-06639> <Skipping Embedded LDAP Server Configuration>
####<Mar 13, 2023 12:51:41 PM> <INFO> <TopologyDiscoverer> <_get_log_filters> <WLSDPLY-06628> <Discovering 1 Log Filters>
####<Mar 13, 2023 12:51:41 PM> <INFO> <TopologyDiscoverer> <_get_log_filters> <WLSDPLY-06629> <Adding Log Filter BEA-090513>
####<Mar 13, 2023 12:51:41 PM> <INFO> <TopologyDiscoverer> <_get_reliable_delivery_policies> <WLSDPLY-06630> <Discovering 0 Reliable Delivery Policies>
####<Mar 13, 2023 12:51:41 PM> <INFO> <TopologyDiscoverer> <_get_virtual_hosts> <WLSDPLY-06647> <Discovering 0 Virtual Hosts>
####<Mar 13, 2023 12:51:41 PM> <INFO> <TopologyDiscoverer> <_get_xml_entity_caches> <WLSDPLY-06632> <Discovering 0 XML Entity Caches>
####<Mar 13, 2023 12:51:41 PM> <INFO> <TopologyDiscoverer> <_get_xml_registries> <WLSDPLY-06634> <Discovering 0 XML Registries>
####<Mar 13, 2023 12:51:41 PM> <INFO> <TopologyDiscoverer> <_get_ws_securities> <WLSDPLY-06649> <Discovering 0 Web Service Securities>
####<Mar 13, 2023 12:51:41 PM> <INFO> <ResourcesDiscoverer> <discover> <WLSDPLY-06300> <Discovering domain model resources>
####<Mar 13, 2023 12:51:41 PM> <INFO> <GlobalResourcesDiscoverer> <get_self_tuning> <WLSDPLY-06441> <Adding Self Tuning>
####<Mar 13, 2023 12:52:00 PM> <INFO> <GlobalResourcesDiscoverer> <get_startup_classes> <WLSDPLY-06442> <Discovering 7 Startup Classes>
####<Mar 13, 2023 12:52:00 PM> <INFO> <GlobalResourcesDiscoverer> <get_startup_classes> <WLSDPLY-06443> <Adding StartupClass AWT Application Context Startup Class>
####<Mar 13, 2023 12:52:00 PM> <INFO> <GlobalResourcesDiscoverer> <get_startup_classes> <WLSDPLY-06443> <Adding StartupClass DMS-Startup>
####<Mar 13, 2023 12:52:00 PM> <INFO> <GlobalResourcesDiscoverer> <get_startup_classes> <WLSDPLY-06443> <Adding StartupClass JRF Startup Class>
####<Mar 13, 2023 12:52:01 PM> <INFO> <GlobalResourcesDiscoverer> <get_startup_classes> <WLSDPLY-06443> <Adding StartupClass ODL-Startup>
####<Mar 13, 2023 12:52:01 PM> <INFO> <GlobalResourcesDiscoverer> <get_startup_classes> <WLSDPLY-06443> <Adding StartupClass SOAStartupClass>
####<Mar 13, 2023 12:52:01 PM> <INFO> <GlobalResourcesDiscoverer> <get_startup_classes> <WLSDPLY-06443> <Adding StartupClass WSM Startup Class>
####<Mar 13, 2023 12:52:01 PM> <INFO> <GlobalResourcesDiscoverer> <get_startup_classes> <WLSDPLY-06443> <Adding StartupClass Web Services Startup Class>
####<Mar 13, 2023 12:52:02 PM> <INFO> <GlobalResourcesDiscoverer> <get_shutdown_classes> <WLSDPLY-06445> <Discovering 1 Shutdown Classes>
####<Mar 13, 2023 12:52:02 PM> <INFO> <GlobalResourcesDiscoverer> <get_shutdown_classes> <WLSDPLY-06446> <Adding ShutdownClass DMSShutdown>
####<Mar 13, 2023 12:52:02 PM> <INFO> <GlobalResourcesDiscoverer> <_get_webapp_container> <WLSDPLY-06615> <Cross Domain Security is enabled, do not remove SecurityConfiguration CredentialEncrypted>
####<Mar 13, 2023 12:52:02 PM> <INFO> <GlobalResourcesDiscoverer> <get_singleton_service> <WLSDPLY-06445> <Discovering 0 Shutdown Classes>
####<Mar 13, 2023 12:52:02 PM> <INFO> <GlobalResourcesDiscoverer> <get_jolt_connection_pool> <WLSDPLY-06449> <Discovering 0 Jolt Connection Pools>
####<Mar 13, 2023 12:52:02 PM> <INFO> <GlobalResourcesDiscoverer> <get_wtc_servers> <WLSDPLY-06451> <Discovering 0 WTC Servers>
####<Mar 13, 2023 12:52:02 PM> <INFO> <CommonResourcesDiscoverer> <get_datasources> <WLSDPLY-06340> <Discovering 12 JDBC System Resources>
####<Mar 13, 2023 12:52:02 PM> <INFO> <CommonResourcesDiscoverer> <get_datasources> <WLSDPLY-06341> <Adding JDBC System Resource EDNDataSource>
####<Mar 13, 2023 12:52:07 PM> <INFO> <CommonResourcesDiscoverer> <get_datasources> <WLSDPLY-06341> <Adding JDBC System Resource EDNLocalTxDataSource>
####<Mar 13, 2023 12:52:11 PM> <INFO> <CommonResourcesDiscoverer> <get_datasources> <WLSDPLY-06341> <Adding JDBC System Resource LocalSvcTblDataSource>
####<Mar 13, 2023 12:52:17 PM> <INFO> <CommonResourcesDiscoverer> <get_datasources> <WLSDPLY-06341> <Adding JDBC System Resource OraSDPMDataSource>
####<Mar 13, 2023 12:52:21 PM> <INFO> <CommonResourcesDiscoverer> <get_datasources> <WLSDPLY-06341> <Adding JDBC System Resource SOADataSource>
####<Mar 13, 2023 12:52:26 PM> <INFO> <CommonResourcesDiscoverer> <get_datasources> <WLSDPLY-06341> <Adding JDBC System Resource SOALocalTxDataSource>
####<Mar 13, 2023 12:52:30 PM> <INFO> <CommonResourcesDiscoverer> <get_datasources> <WLSDPLY-06341> <Adding JDBC System Resource WLSSchemaDataSource>
####<Mar 13, 2023 12:52:34 PM> <INFO> <CommonResourcesDiscoverer> <get_datasources> <WLSDPLY-06341> <Adding JDBC System Resource mds-owsm>
####<Mar 13, 2023 12:52:39 PM> <INFO> <CommonResourcesDiscoverer> <get_datasources> <WLSDPLY-06341> <Adding JDBC System Resource mds-soa>
####<Mar 13, 2023 12:52:43 PM> <INFO> <CommonResourcesDiscoverer> <get_datasources> <WLSDPLY-06341> <Adding JDBC System Resource opss-audit-DBDS>
####<Mar 13, 2023 12:52:47 PM> <INFO> <CommonResourcesDiscoverer> <get_datasources> <WLSDPLY-06341> <Adding JDBC System Resource opss-audit-viewDS>
####<Mar 13, 2023 12:52:52 PM> <INFO> <CommonResourcesDiscoverer> <get_datasources> <WLSDPLY-06341> <Adding JDBC System Resource opss-data-source>
####<Mar 13, 2023 12:52:56 PM> <INFO> <CommonResourcesDiscoverer> <get_foreign_jndi_providers> <WLSDPLY-06342> <Discovering 0 Foreign JNDI Providers>
####<Mar 13, 2023 12:52:56 PM> <INFO> <CommonResourcesDiscoverer> <get_mail_sessions> <WLSDPLY-06344> <Discovering 1 Mail Sessions>
####<Mar 13, 2023 12:52:56 PM> <INFO> <CommonResourcesDiscoverer> <get_mail_sessions> <WLSDPLY-06345> <Adding Mail Session soaMailSession>
####<Mar 13, 2023 12:52:56 PM> <SEVERE> <discover> <main> <WLSDPLY-20035> <discoverDomain encountered an unexpected runtime exception.  Please file an issue on GitHub and attach the log file and stdout. Exception: exceptions.AttributeError>

Issue Log for discoverDomain version 3.0.1 running WebLogic version 12.2.1.4.0.221210 online mode:

SEVERE Messages:

        1. WLSDPLY-20035: discoverDomain encountered an unexpected runtime exception.  Please file an issue on GitHub and attach the log file and stdout. Exception: exceptions.AttributeError

Total:   SEVERE :    1  WARNING :    0

discoverDomain.sh failed (exit code = 2)

```

Regards,
Piotr Michalski 
Oracle ACS

**Comment by Michalski-Piotr** [comment]
[discoverDomain.log](https://github.com/oracle/weblogic-deploy-tooling/files/10957628/discoverDomain.log)

**Comment by robertpatrick** [comment]
This fix will be included in WDT 3.0.3, which we hope to release in the next week or so.
